build.gradle

```diff
@@ -24,19 +24,63 @@ plugins {
 
 allprojects {
     repositories {
-        google()
-        jcenter()
-
         maven {
             url "https://maven.mozilla.org/maven2"
+            content {
+                // We want to fetch all our known Mozilla groups from here.
+                includeGroup GroupMatching.GROUP_MOZILLA_APPSERVICES
+                includeGroup GroupMatching.GROUP_MOZILLA_COMPONENTS
+                includeGroup GroupMatching.GROUP_MOZILLA_GECKOVIEW
+            }
+        }
+
+        google() {
+            content {
+                // We want to fetch all AndroidX, Google and Firebase groups from here.
+                includeGroupByRegex GroupMatching.REGEX_ANDROIDX
+                includeGroupByRegex GroupMatching.REGEX_ANDROID
+                includeGroupByRegex GroupMatching.REGEX_GOOGLE_ANDROID
+                includeGroup GroupMatching.GROUP_GOOGLE_FIREBASE
+
+                // We never want to fetch Mozilla groups from here.
+                excludeGroupByRegex GroupMatching.REGEX_MOZILA
+            }
         }
 
-        ////////////////////////////////////////////////////////////////////////////////////////////
-        // Servo
-        ////////////////////////////////////////////////////////////////////////////////////////////
+        mavenCentral() {
+            // We list Maven Central before jcenter since we prefer pulling artifacts from there if available. Maven
+            // Central has a more strict vetting process when adding new artifacts.
+
+            content {
+                // We never want to fetch Mozilla, AndroidX, Google or Firebase groups from Maven Central.
+                excludeGroupByRegex GroupMatching.REGEX_MOZILA
+                excludeGroupByRegex GroupMatching.REGEX_ANDROIDX
+                excludeGroupByRegex GroupMatching.REGEX_ANDROID
+                excludeGroup GroupMatching.GROUP_GOOGLE_FIREBASE
+
+                // We are not excluding REGEX_GOOGLE_ANDROID here since there's one group that is not getting published
+                // on maven.google.com so far (but on Maven Central, also jcenter but we prefer Maven Central):
+                // com.google.android.apps.common.testing.accessibility.framework
+                // See: https://github.com/google/Accessibility-Test-Framework-for-Android/issues/14
```

Comment:
This exposes you to vulnerabilities on com.google.android dependencies. I'd prefer if you explicitly excluded this dependency from the matching like we do on FFTV: https://searchfox.org/mozilla-mobile/source/firefox-tv/buildSrc/src/main/java/org/mozilla/gradle/Dependencies.kt#28 Though I understand the trade-off of complexity.

Reply:
Yes, I think they should be escaped and yes, I think it should be framework. Nice catch.

Reply:
fwiw, we could probably share these declarations with our android gradle plugin...

Reply:
I'm not sure it's worth the time though.